js/whitebit.js

```diff
@@ -974,64 +974,65 @@ module.exports = class whitebit extends Exchange {
          * @param {object} params extra parameters specific to the whitebit api endpoint
          * @returns {object} an [order structure]{@link https://docs.ccxt.com/en/latest/manual.html#order-structure}
          */
-        let method = undefined;
         await this.loadMarkets ();
         const market = this.market (symbol);
         const request = {
             'market': market['id'],
             'side': side,
             'amount': this.amountToPrecision (symbol, amount),
         };
-        const stopPrice = this.safeNumber2 (params, 'stopPrice', 'activationPrice');
-        if (stopPrice !== undefined) {
-            // it's a stop order
+        const isLimitOrder = (type === 'limit');
+        const isMarketOrder = (type === 'market');
+        const stopPrice = this.safeNumber2 (params, 'triggerPrice', 'stopPrice');
+        const isStopOrder = (stopPrice !== undefined);
+        const timeInForce = this.safeString (params, 'timeInForce');
+        const fok = (timeInForce === 'FOK');
+        if (timeInForce === 'IOC' || timeInForce === 'PO') {
```

Comment:
If we only want to accept a single TIF value, first I think we could probably make it default so the user would not have to explicitly declare it, making things easier on the userland.
```
const timeInForce = this.safeString (params, 'timeInForce', 'FOK');
````
Second, if we really want to accept 'FOK' only we should make a whitelist filtering instead of blacklisting all the possibilities. Meaning, the user can virtually send anything as the TIF parameter and we can't have infinite if conditions checking all the values like `if (tif === 'a' || tif === 'b' || tif === 'GOK' || ...) { throw (...)}`

I would say we could simplify it by doing this:
```Javascript
const timeInForce = this.safeString (params, 'timeInForce');
const fok = (timeInForce === 'FOK');
id (!fok) {
    throw(...)
} 
```
do you agree?

Reply:
I Agree with the whitelisting methodology, will add it now, but I dont think we should be defaulting every users order to FOK? Or am I misunderstanding what you mean with 
`const timeInForce = this.safeString (params, 'timeInForce', 'FOK');`